### ISSUE TITLE
Finding incorrect 'libpython'

### ISSUE THREAD

**Issue by rosskarchner** [python-pillow/Pillow#1759]
I'm on RHEL6. there are two different copies of Python 2.7 installed:
- a compiled-from-source Python 2.7.6 in /usr/local
- Python 2.7 from [software collection](https://www.softwarecollections.org/en/scls/rhscl/python27/)

When I attempt to build pillow, with the software collection activated...

```
# which python
/opt/rh/python27/root/usr/bin/python

# pkg-config --libs --cflags  python                                                                            
-I/opt/rh/python27/root/usr/include/python2.7  -L/opt/rh/python27/root/usr/lib64 -lpython2.7  

# env | grep LD
LD_LIBRARY_PATH=/opt/rh/python27/root/usr/lib64
```

... it seems to be trying to link with the other libpython (and failing, since it's not shared):

```
/usr/bin/ld: /usr/local/lib/libpython2.7.a(abstract.o): relocation R_X86_64_32 against `.rodata.str1.8' can not be used when making a shared object; 
recompile with -fPIC
/usr/local/lib/libpython2.7.a: could not read symbols: Bad value
```

Is there a way to ensure that while pillow is building, it uses the the version of libpython I want (/opt/rh/python27/root/usr/lib64/libpython2.7.so)?

**Comment by rosskarchner** [comment]
I did a little spelunking, and interestingly the /opt/rh/python27/root/usr/lib64 directory ends up as the last item on the search path:

```python
self.compiler.library_dirs
['/opt/rh/python27/root/usr/lib', '/lib64', '/usr/lib64', '/usr/local/lib', '/usr/lib', '/opt/rh/python27/root/usr/lib64']
```

**Comment by rosskarchner** [comment]
OK, I was able to work around this with:

```
export LD_RUN_PATH=$LD_LIBRARY_PATH
```

I wonder if LD_LIBRARY_PATH should be one of the variables checked on [this line](https://github.com/python-pillow/Pillow/blob/master/setup.py#L185)

**Comment by radarhere** [comment]
Just a note - line 185 from the previous commit is now line 276. To make understanding this easier, the line is `for k in ('LD_RUN_PATH', 'LIBRARY_PATH', 'LIB'):`

**Comment by radarhere** [comment]
Googling for information around LD_LIBRARY_PATH, there are a lot of warnings around using this. Here is one such warning - https://medium.com/macos-is-not-linux-and-other-nix-reflections/d-o-y-ou-ld-library-path-you-6ab0a6135a33
> In fact, literature over the HiddenWasp malware, considered a game-changer for Linux vulnerabilities, suggests to be suspicious of a compromise if any system has LD_LIBRARY_PATH set³:

**Comment by radarhere** [comment]
Also, note that Pillow no longer supports Python 2.7.

**Comment by radarhere** [comment]
What was the command used to compile Pillow?

Looking in that software collection, https://www.softwarecollections.org/en/scls/user/rhscl/?search=python, its highest version is Python 3.6, which we're about to drop support for.

**Comment by radarhere** [comment]
Pillow 9.0 has now been released, dropping support for Python 3.6.

Since https://www.softwarecollections.org/en/scls/user/rhscl/?search=python doesn't support anything higher, I'm going to close this. Let us know if you have any further thoughts.
